docs/learn/learn-guides-coretime-swap.md

```diff
@@ -0,0 +1,102 @@
+
+In order to check if a parachain is locked or not, teams need to query `registrar.paras` with the
+corresponding paraID on the relay chain. It's important to note that since
+[RFC #14](https://github.com/polkadot-fellows/RFCs/pull/14), if a parachain never produced a block
+the it's by default unlocked.
+
+## Initiate Swap from the Parachain
+
+To swap from a parachain, an XCM message needs to be sent from the parachain to the relay chain.
+This XCM message must be a send instruction that wraps the `registrar.swap` extrinsic that needs to
+be executed on the relay chain.
+
+Below, you can find an example hex encoded call for Composable Finance, assuming the team is still
+operating under SUDO.
+
+[0x0200290003010003140004000000000700e876481713000000000700e87648170006000700e40b540282380100284603e3070000320d0000140d01000001008d1f](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fcomposable-rpc.dwellir.com#/extrinsics/decode/0x0200290003010003140004000000000700e876481713000000000700e87648170006000700e40b540282380100284603e3070000320d0000140d01000001008d1f)
+
+As for weights (size and time), there's an overestimation from what can be queried from the polkadot
+relay chain.
+
+To check the exact weights, you can navigate to Polkadot JS UI > Developer > Runtime Calls and then
+select `transactionPaymentCallInfo` and `queryCallInfo` and include the extrinsic being sent in the
+`send` instruction of the XCM message:
+[0x4603e3070000320d0000](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.ibp.network%2Fpolkadot#/extrinsics/decode/0x4603e3070000320d0000)
+
+![TX-payment-API-Lease-Swap](../assets/coretime/tx-payment-api-weights-lease-swap.png)
+
+## Perform Swap on the Relay chain for an unlocked Parachain
+
+In the case a parachain is unlocked, then the manager of the parachain on the relaychain must send
+the extrinsic directly on the relaychain.
+
+In the case of paraID 3378 wanting to swap with Composable Finance, the extrinsic to send from its
+manager account `13B8Tdhi4EuruwzVt3gdcTopXpqbUzmncutiQKKNbd8cpU91` is:
+
+[0x4603320d0000e3070000](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.ibp.network%2Fpolkadot#/extrinsics/decode/0x4603320d0000e3070000)
+
+## Coretime Swap Prerequisites
+
+We need to ensure the parachains have a valid cores to execute the swap instruction. This can be
+achieved either by getting a core in bulk from the coretime system chain, or by leveraging the
+onDemand functionality on relay chain. Assuming that teams have a working parachain (with a valid
+core) and a stalled parachain, the following instructions explain how to get a core `onDemand` for
+the stalled parachain.
+
+### Swap through an OnDemand Core
+
+This functionality exists on the relay chain and allows users to request for 1 block of validation
+of their parachain. In order to place an order for on demand, any account must call the
+`onDemand.placeOrderAllowDeath` extrinsic.
+
+![onDemand-block-production](../assets/coretime/onDemand-block-production.png)
```

Comment:
Only issue is what paraID to include, maybe we can add something like "your_para_id". And my OCD would also say with an updated metadata :)

Reply:
I was thinking about the composable call included above, I think since that case is used up there we can use it throughout, or alternatively just have some fictitious swap between asset hub and bridge hub or something so people can see which numbers go where

Reply:
Fair point. I'll go along with the composable example. This would be 2019 (the chain that is going to send the XCM call)